src/services/formatting/smartIndenter.ts

```diff
@@ -531,9 +531,23 @@ namespace ts.formatting {
             return false;
         }
 
-        export function nodeWillIndentChild(parent: TextRangeWithKind, child: TextRangeWithKind | undefined, indentByDefault: boolean): boolean {
+        export function nodeWillIndentChild(parent: TextRangeWithKind, child: Node | undefined, indentByDefault: boolean): boolean {
             const childKind = child ? child.kind : SyntaxKind.Unknown;
+
             switch (parent.kind) {
+                case SyntaxKind.VariableDeclaration:
+                case SyntaxKind.PropertyAssignment:
+                case SyntaxKind.ObjectLiteralExpression:
+                    if (childKind === SyntaxKind.ObjectLiteralExpression) {
+                        const sourceFile = child.getSourceFile();
+                        if (sourceFile) {
+                            // May not be defined for synthesized nodes.
+                            const startLine = sourceFile.getLineAndCharacterOfPosition(child.getStart()).line;
+                            const endLine = sourceFile.getLineAndCharacterOfPosition(child.getEnd()).line;
+                            return startLine === endLine;

```

Comment:
Did any of the tests cover the case where this is false?

Reply:
I've added test where the startline is the same as the endline, despite leading trivia making the object-literal child's `pos` inappropriate to refer to.